webrender/src/render_backend.rs

```diff
@@ -1079,6 +1085,13 @@ impl RenderBackend {
             self.result_tx.send(msg).unwrap();
             profile_counters.reset();
             doc.render_on_hittest = false;
+        } else if op.render {
+            // WR-internal optimization to avoid doing a bunch of render work if
+            // there's no pixels. We still want to pretend to render and request
+            // a composite to make sure that the callbacks (particularly the
+            // new_frame_ready callback below) has the right flags.
+            let msg = ResultMsg::PublishPipelineInfo(doc.get_updated_pipeline_info());
```

Comment:
It's duplicating the way Renderer gets the pipeline updates. Perhaps, we can remove this from `PublishDocument` one and always make it so the `PublishPipelineInfo` is sent beforehand?